### PR TITLE
TST/DOC : Unit tests for `pysat.utils.generate_instrument_list` and docstring scrub for utils area

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #857 and #860 (partial)

- Adds unit tests for `pysat.utils.generate_instrument_list`.
- Adjusts class-level definitions of variables in tests
- Docstring scrub:
  - instrument_test_class
  - utils and submodules
  - test_utils and submodules

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

flake8-docstrings

**Test Configuration**:
* Operating system: Mac OS X 10.15.7
* Version number: Python 3.8.2
* flake8, flake8-docstrings

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes (part of existing changelog)

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/tests/test_utils_coords.py [hunk 1]
```diff
@@ -94,18 +94,24 @@ def test_bad_lon_name_update_longitude(self):
 
 
 class TestCalcSLT():
-    """Unit tests for `calc_solar_local_time` and integration tests for
-    `update_longitude`.
+    """Unit tests for `calc_solar_local_time`.
+
+    Note
+    ----
+    Includes integration tests with `update_longitude`.
```
Comment (jklenzing) [inline pysat/tests/test_utils_coords.py:101]: This was too long to include with above, unless with abbreviate something

File: pysat/tests/test_utils_files.py [hunk 2]
```diff
@@ -21,10 +20,11 @@
 
 class TestParseDelimitedFilenames():
     """Unit tests for the `parse_delimited_filename` function."""
-    temporary_file_list = False
 
     def setup(self):
-        """Runs before every method to create a clean testing setup."""
+        """Set up the unit test environment for each method."""
+
+        self.temporary_file_list = False
```
Comment (jklenzing) [inline pysat/tests/test_utils_files.py:27]: Not using inheritance here, so I moved the definition to method-level setup.  We could define this as a class-level setup if we plan to expand this to repeat tests with `temporary_file_list=True`.

File: pysat/tests/test_utils.py [hunk 3]
```diff
@@ -824,3 +826,64 @@ def test_without_fail(self):
...
+
+        tags = [inst['tag'] for inst in self.inst_list[list_name]]
+        assert ('no_download' in tags) == output
+        return
+
+    def test_user_info_pass_through(self):
+        """Test that user info passes through to correct instruments."""
+
+        for inst in self.inst_list['download']:
+            # user_info should only be in pysat_testmodel
```
Comment (aburrell) [inline pysat/tests/test_utils.py:880]:
```suggestion
            # `user_info` should only be in `pysat_testmodel`
```

File: pysat/tests/test_utils.py [hunk 4]
```diff
@@ -824,3 +826,64 @@ def test_without_fail(self):
...
+            assert (('user_info' in inst.keys())
+                    == ('pysat_testmodel' in str(inst['inst_module'])))
+            if 'user_info' in inst.keys():
+                # User info should be correct
+                assert inst['user_info'] == self.user_info['pysat_testmodel']
+        for inst in self.inst_list['no_download']:
+            # user_info should not be in any of these
```
Comment (aburrell) [inline pysat/tests/test_utils.py:887]:
```suggestion
            # `user_info` should not be in any of these
```

File: pysat/utils/__init__.py [hunk 5]
```diff
@@ -1,7 +1,5 @@
 # -*- coding: utf-8 -*-
-"""
-pysat.utils - utilities for running pysat
-=========================================
+"""utilities for running pysat.
```
Comment (aburrell) [inline pysat/utils/__init__.py:2]:
```suggestion
"""Utilities supporting pysat classes, packages, and the testing environment.
```

File: pysat/utils/_core.py [hunk 6]
```diff
@@ -117,7 +117,7 @@ def scale_units(out_unit, in_unit):
 
 
 def listify(iterable):
-    """Returns a flattened list of iterable if not already a list
+    """Return a flattened list of iterable if not already a list.
```
Comment (aburrell) [inline pysat/utils/_core.py:120]:
```suggestion
    """Produce a flattened list of items from input that may not be iterable.
```

File: pysat/utils/time.py [hunk 7]
```diff
@@ -3,9 +3,7 @@
 # Full author list can be found in .zenodo.json file
 # DOI:10.5281/zenodo.1199703
 # ----------------------------------------------------------------------------
-"""
-pysat date and time utilities
-"""
+"""pysat date and time utilities."""
```
Comment (aburrell) [inline pysat/utils/time.py:6]:
```suggestion
"""Date and time handling utilities."""
```

File: pysat/utils/time.py [hunk 8]
```diff
@@ -316,7 +323,7 @@ def create_datetime_index(year=None, month=None, day=None, uts=None):
 
 
 def filter_datetime_input(date):
-    """Returns datetime that only includes year, month, and day.
+    """Return datetime that only includes year, month, and day.
```
Comment (aburrell) [inline pysat/utils/time.py:326]:
```suggestion
    """Create a datetime object that only includes year, month, and day.
```

File: pysat/utils/time.py [hunk 9]
```diff
@@ -360,7 +367,7 @@ def filter_datetime_input(date):
 
 
 def today():
-    """Returns today's date (UTC), with no hour, minute, second, etc.
+    """Return today's date (UTC), with no hour, minute, second, etc.
```
Comment (aburrell) [inline pysat/utils/time.py:370]:
```suggestion
    """Obtain today's date (UTC), with no hour, minute, second, etc.
```

**Review by rstoneback — APPROVED** [review]
Thanks @jklenzing. Looks good to me.

**Comment by jklenzing** [comment]
Thanks @rstoneback.  Going to see if @aburrell has any further comments before I merge.

File: pysat/utils/time.py [hunk 10]
```diff
@@ -214,14 +212,24 @@ def freq_to_res(freq):
...
+    ----------
+    start : dt.datetime or list-like of dt.datetime
+        The beginning of the date range.  Supports list, tuple, or ndarray of
+        start dates.
+    stop : dt.datetime or list-like of dt.datetime
+        The end of the date range.  Supports list, tuple, or ndarray of
+        stop dates.
+    freq : str
+        The frequency of the desired output.  Codes correspond to pandas
+        date_range codes: D daily, M monthly, S secondly
```
Comment (aburrell) [inline pysat/utils/time.py:227]:
```suggestion
        date_range codes: 'D' daily, 'M' monthly, 'S' secondly
```

**Review by aburrell — COMMENTED** [review]
Only found one thing!

**Review by aburrell — APPROVED** [review]

**Pull request merged**